jung-algorithms/src/main/java/edu/uci/ics/jung/algorithms/shortestpath/MinimumSpanningTree.java

```diff
@@ -111,14 +111,23 @@
       if (connectedNode == null) {
         tree.addNode(node);
       } else {
-        tree.putEdgeValue(node, connectedNode, graph.edgeValue(node, connectedNode));
+        tree.putEdgeValue(
+            node,
+            connectedNode,
+            graph
+                .edgeValue(node, connectedNode)
+                .orElseThrow(() -> new IllegalStateException("this is a bug in JUNG")));
```

Comment:
Suggest changing the text to "unexpected exception caused by bug in graph data structures", or similar.

Reply:
Done!